### PR TITLE
Display figures on a new github pages site.

### LINKED ISSUE #46: Push Algorithm Analysis result plots to a static github page
### Feature description

The Algorithm Analysis workflow automatically generates some nice plots. It may be possible to automatically push those to a static github page so they're easy to view.

### Describe the solution

The [Figures](https://github.com/OSIPI/TF2.4_IVIM-MRI_CodeCollection/actions/runs/8098690315/artifacts/1286493783) generated in the [Algorithm Analysis workflow](https://github.com/OSIPI/TF2.4_IVIM-MRI_CodeCollection/actions/runs/8098690315) could probably be uploaded to a simple static github page. This would be a nice way to visualize the most recent results.

### Describe alternatives

_No response_

### Additional context

I am not sure if it is possible to fully automate this. It would involve exploring the options and testing it. If it's not possible, that's a good finding as well.

### Are you working on this?

None

### PR DESCRIPTION
**Pull request by AhmedBasem20**
### Describe the changes you have made in this PR

Displayed the generated plots from the analysis workflow to a new github-pages site.
### Results
Example from my fork: https://ahmedbasem20.github.io/TF2.4_IVIM-MRI_CodeCollection/

@etpeterson To enable this on this repository we need the following steps:

1. create a new branch named `gh-pages`.
2. From the pages section on the repository settings, set `gh-pages` as a source branch.
image

### Link this PR to an issue [optional]

Fixes #46

### CODE REVIEW

File: .github/workflows/analysis.yml [hunk 1]
```diff
@@ -146,6 +146,20 @@ jobs:
               durations.pdf
               curve_plot.pdf
               fitted_curves.pdf
+      - name: Download artifacts
```
Comment (etpeterson) [inline .github/workflows/analysis.yml:149]: Is this download step necessary? The figures should already be present

File: index.html [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
```
Comment (etpeterson) [inline index.html:1]: Perhaps move this into a website folder?

File: index.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    </style>
+</head>
+<body>
+    <h1>Algorithm Analysis Figures</h1>
+    <embed src="curve_plot.pdf" type="application/pdf" frameborder="0" scrolling="auto">
+    <embed src="D_limited.pdf" type="application/pdf" frameborder="0" scrolling="auto">
+    <embed src="D.pdf" type="application/pdf" frameborder="0" scrolling="auto">
+    <embed src="Dp_limited.pdf" type="application/pdf" frameborder="0" scrolling="auto">
+    <embed src="Dp.pdf" type="application/pdf" frameborder="0" scrolling="auto">
+    <embed src="durations.pdf" type="application/pdf" frameborder="0" scrolling="auto">
```
Comment (etpeterson) [inline index.html:30]: Probably move curve and duration to the bottom as they're probably less interesting.

File: .github/workflows/analysis.yml [hunk 4]
```diff
@@ -147,6 +147,14 @@ jobs:
               curve_plot.pdf
               fitted_curves.pdf
 
+      - name: Create Build Directory and Move PDFs
+        run: mv *.pdf website/
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@4.1.1
```
Comment (etpeterson) [inline .github/workflows/analysis.yml:153]: How does this differ from [this](https://github.com/actions/upload-pages-artifact)? Using the official one seems to be the safer bet, if possible.

**Comment by etpeterson** [comment]
This all looks good but I'm wondering how this actually works on the repository. We just went through a big cleaning process and we don't want to have to deal with that again, so I'm hyper vigilant about uploading things to the repository. It _seems_ like this pushes the pdfs to the repository each time. I see [this](https://www.reddit.com/r/rust/comments/wy3j50/psa_if_youre_using_ghpages_to_host_your/) discussion about that, for example. That then points to [this](https://github.com/actions/deploy-pages) as a way to push but not keep the history.

In short, I think whatever we do, we don't want the history of the pdfs pushed, just the most recent.

**Review by etpeterson — CHANGES_REQUESTED** [review]

**Comment by AhmedBasem20** [comment]
Thanks @etpeterson for the review, I used the deployment method you suggested and it worked fine, see: https://github.com/AhmedBasem20/TF2.4_IVIM-MRI_CodeCollection/actions/runs/8139256741

**Comment by etpeterson** [comment]
> Thanks @etpeterson for the review, I used the deployment method you suggested and it worked fine, see: https://github.com/AhmedBasem20/TF2.4_IVIM-MRI_CodeCollection/actions/runs/8139256741

This run only ran 4 algorithms but I see on the hosted site that all of them ran. Why this discrepancy? I don't see anything in the code that should be changing that.

**Comment by AhmedBasem20** [comment]
My bad. I wanted to see the results quickly, so I removed most of the algorithms on this workflow. But the workflow of the committed code on this pull request works fine: https://github.com/AhmedBasem20/TF2.4_IVIM-MRI_CodeCollection/actions/runs/8139531990 

> I also see there is a few pushes that caused commits, but I don't see any changes more recently. I'm not sure how this should act, but with the algorithm discrepancy I'm still hesitant to merge until it's all clear.


Those pushes caused by the old action that deploys to `gh-pages`. It is removed now.

File: .github/workflows/analysis.yml [hunk 5]
```diff
@@ -6,6 +6,11 @@ on:
       - 'main'
```
Comment (etpeterson) [inline .github/workflows/analysis.yml:6]: This workflow runs on `main` and anything that starts with `analysis`. We probably only want the website to update from the `main` branch, so can we run it but not upload if it's running an `analysis` branch?

**Review by etpeterson — CHANGES_REQUESTED** [review]
This looks good, just one small comment. It is a little opaque to me how the publishing works though. Have you found a good resource?

Perhaps another change would be however we should structure the website. I'm imagining a base page and we link to the results from there, or perhaps they're on the base page somewhere. Right now it's just the results page, correct? Maybe the comment in the related issue will get answered to help here.

**Comment by AhmedBasem20** [comment]
@etpeterson the publishing action depends on the artifact we uploaded, I used [this action](https://github.com/actions/starter-workflows/blob/main/pages/static.yml) as a reference.
Will the documentation page be on this website? If so, we build this page along with the docs and add it there. But for now, I can improve the layout of the current HTML page until we get the documentation.

**Comment by etpeterson** [comment]
This looks good. I think there's an open question about where the results should land and if the index.html page should start with a little more on there, but that's for @petravanhoudt and @oliverchampion.

**Comment by etpeterson** [comment]
Just following up @oliverchampion or @petravanhoudt if you have ideas on the structure of the page.

**Comment by AhmedBasem20** [comment]
superseded by #59
